### DISCUSSION TITLE
progress bar from progressr does not show when running pipeline

### DISCUSSION THREAD

**Discussion by grantn5** [ropensci/targets#1111] [Help]
when writing a function with the `progressr` package the progress bar does not show when running `tar_make()`

REPREX

```
library(targets)

tar_option_set(packages = c("progressr"))

slow_sum <- function(x) {
  p <- progressr::progressor(along = x)
  sum <- 0
  for (kk in seq_along(x)) {
    Sys.sleep(0.1)
    sum <- sum + x[kk]
    p(message = sprintf("Adding %g", x[kk]))
  }
  sum
}

list(
  tar_target(with_progress(y <- slow_sum(1:10)))
)

```
Example is taken from 
https://cran.r-project.org/web/packages/progressr/vignettes/progressr-intro.html

How would I get the bar to show

**Comment by wlandau** [comment]
@grantn5, I am not very familiar with `progressr`, but this issue seems to be related to how the progress bar is configured. Using a simpler pipeline I see messages like "M: added value 1":

```r
# _targets.R file
targets::tar_target(y, progressr::with_progress(progressr::slow_sum(1:10)))
```

On the flip side, if `slow_sum()` is a custom function inside a `callr` process, then the messages do not appear. This matters because `targets` uses `callr` behind the scenes, and if that is the source of the problem, then this is not an issue with `targets`.

```r
callr::r(
  func = function() {
    custom_slow_sum <- function(x) {
      p <- progressr::progressor(along = x)
      sum <- 0
      for (kk in seq_along(x)) {
        Sys.sleep(0.1)
        sum <- sum + x[kk]
        p(message = sprintf("Adding %g", x[kk]))
      }
      sum
    }
    progressr::with_progress(custom_slow_sum(1:10))
  },
  show = TRUE
)
```

**Reply by grantn5** [reply]
Thank you very much for the quick response,  

I have had a quick look at `callr` and I am not sure it is possible to display the bar in the current targets framework, as you say when you wrap the `progressr` function in a `callr` function the bar is not shown.
